cpp/include/Ice/EncodingVersion.h

```diff
@@ -0,0 +1,38 @@
+//
+// Copyright (c) ZeroC, Inc. All rights reserved.
+//
+
+#ifndef ICE_ENCODING_VERSION_H
+#define ICE_ENCODING_VERSION_H
+
+#include "Config.h"
+#include "Comparable.h"
+#include <cstdint>
+
+namespace Ice
+{
+    /**
+     * Represents a version of the Slice encoding.
+     * \headerfile Ice/Ice.h
+     */
+    struct EncodingVersion
```

Comment:
The "hand generated" version, but without the marshaling/unmarshaling support to avoid circular dependencies.

The marshaling/unmarshaling support for EncodingVersion is not at the bottom of StreamHelpers.h